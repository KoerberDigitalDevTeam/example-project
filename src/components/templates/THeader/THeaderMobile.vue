<template>
  <div class="theader-mobile--sticky">
    <div class="theader-mobile">
      <AButton
        v-testId.theader.mobile-button
        :tertiary="true"
        icon
        color="dWhite"
        class="theader-mobile__button mr-6"
        data-testid="btn-mob-header-open"
        @click.stop="handleToggleSidebar"
      >
        <v-icon>mdi-menu</v-icon>
      </AButton>

      <v-list-item-title v-testId.theader.mobile-title class="theader-mobile__title">
        <h4>{{ title }}</h4>
      </v-list-item-title>
    </div>
  </div>
</template>

<script>
import { mapState, mapActions } from 'vuex';
import AButton from '@/components/atoms/AButton/AButton.vue';

export default {
  name: 'THeaderMobile',
  components: {
    AButton,
  },

  props: {
    title: {
      type: String,
      required: true,
    },
  },

  data() {
    return {
      isSidebarOpen: false,
    };
  },

  computed: {
    ...mapState({
      isSidebarOpenGlobal: (state) => state.global.isSidebarOpen,
    }),
  },

  watch: {
    isSidebarOpenGlobal(newValue) {
      this.isSidebarOpen = newValue;
    },
  },

  methods: {
    ...mapActions({ toggleSidebar: 'global/toggleSidebar' }),
    handleToggleSidebar() {
      this.isSidebarOpen = !this.isSidebarOpen;
      this.toggleSidebar(this.isSidebarOpen);
    },
  },
};
</script>

<style lang="scss" scoped>
.theader-mobile {
  display: flex;
  background: $d-black;
  color: $d-white;
  padding: 4px 0 4px 20px;
  overflow: hidden;
}

.theader-mobile--sticky {
  position: sticky;
  top: 0;
  z-index: 4;
}
</style>
