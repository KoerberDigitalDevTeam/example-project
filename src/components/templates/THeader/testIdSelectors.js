export default {
  title: 'header-title',
  toolbar: 'header-toolbar',
  toolbarLeft: 'header-toolbar-left',
  toolbarRight: 'header-toolbar-right',
};
