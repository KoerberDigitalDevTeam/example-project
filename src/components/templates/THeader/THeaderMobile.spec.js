import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import testIdSelectors from '@/components/templates/THeader/testIdSelectors';
import THeaderMobile from '@/components/templates/THeader/THeaderMobile.vue';

function createWrapper(options = {}) {
  const localVue = createLocalVue();

  const global = {
    namespaced: true,
    state: {
      isSidebarOpen: false,
    },
    actions: {
      toggleSidebar: jest.fn(),
    },
  };

  localVue.use(Vuex);

  const store = new Vuex.Store({
    modules: {
      global,
    },
  });

  return shallowMount(THeaderMobile, {
    localVue,
    store,
    vuetify: new Vuetify(),
    ...options,
  });
}

describe('THeaderMobile', () => {
  describe('Template:', () => {
    it('Should render THeaderMobile', () => {
      const wrapper = createWrapper({
        propsData: {
          title: 'Test title',
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
    ~it('Should have the title inside the <h4>', () => {
      const wrapper = createWrapper({
        propsData: {
          title: 'Test title',
        },
      });
      const element = wrapper.find(`[data-testid=theader-mobile-title]`);

      expect(element.text()).toBe('Test title');
    });
  });

  describe('Props Interface:', () => {
    it('Should have title prop required and must be a string', () => {
      expect(THeaderMobile.props).toMatchObject({
        title: {
          type: String,
          required: true,
        },
      });
    });
  });

  describe('Watch:', () => {
    it("Should isSidebarOpenGlobal set 'isSidebarOpen' with 'newValue'", () => {
      const localThis = {
        isSidebarOpen: false,
      };

      const payload = true;

      THeaderMobile.watch.isSidebarOpenGlobal.call(localThis, payload);

      expect(localThis.isSidebarOpen).toBe(payload);
    });
  });

  describe('Methods:', () => {
    it("Should handleToggleSidebar invert 'isSidebarOpen' value and call action 'toggleSidebar'", () => {
      const localThis = {
        isSidebarOpen: false,
        toggleSidebar: jest.fn(),
      };

      THeaderMobile.methods.handleToggleSidebar.call(localThis);

      expect(localThis.isSidebarOpen).toBeTruthy();
      expect(localThis.toggleSidebar).toHaveBeenCalledWith(localThis.isSidebarOpen);
    });
  });
});
