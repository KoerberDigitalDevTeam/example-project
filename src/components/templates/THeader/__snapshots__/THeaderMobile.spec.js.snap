// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`THeaderMobile Template: Should render THeaderMobile 1`] = `
<div class="theader-mobile--sticky">
  <div class="theader-mobile">
    <abutton-stub primary="true" tertiary="true" color="dWhite" icon="" data-testid="theader-mobile-button" class="theader-mobile__button mr-6">
      <v-icon-stub>mdi-menu</v-icon-stub>
    </abutton-stub>
    <v-list-item-title-stub class="theader-mobile__title" data-testid="theader-mobile-title">
      <h4>Test title</h4>
    </v-list-item-title-stub>
  </div>
</div>
`;
